src/pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -452,6 +452,18 @@ def log(
     ) -> None:
         """See :meth:`~pytorch_lightning.core.module.LightningModule.log`"""
         # no metrics should be logged with graphs
+        if (
+            sync_dist
+            and not isinstance(value, Tensor)
+            and (isinstance(value, Metric) or any(isinstance(v, Metric) for v in value.values()))
+        ):
+            raise MisconfigurationException(
```

Comment:
I would personally put this in the `LightningModule.log` method where we have all the other validation checks